test_common.py

```diff
@@ -0,0 +1,219 @@
+    "jaccard":_jaccard ,
+    "consensus_score":consensus_score    
+    }
+      
+SUPERVISED_METRICS = {
+    "adjusted_mutual_info_score":adjusted_mutual_info_score ,
+    "adjusted_rand_score":adjusted_rand_score,
+    "completeness_score":completeness_score,
+    "contingency_matrix":contingency_matrix,
+    "homogeneity_score":homogeneity_score,
+    "entropy":entropy,
+    "expected_mutual_information":expected_mutual_information,
+    "homogeneity_completeness_v_measure":homogeneity_completeness_v_measure,
+    "mutual_info_score":mutual_info_score,
+    "normalized_mutual_info_score":normalized_mutual_info_score,
+    "v_measure_score":v_measure_score
+    }
+        
+UNSUPERVISED_METRICS={
+    "silhouette_score":silhouette_score,
+    "silhouette_samples":silhouette_samples
+    }
+        
+ALL_METRICS = dict()
+ALL_METRICS.update(BICLUSTER_METRICS)
+ALL_METRICS.update(SUPERVISED_METRICS)
+ALL_METRICS.update(UNSUPERVISED_METRICS)
+
+# Lists of metrics with common properties
+# ---------------------------------------
+# Lists of metrics with common properties are used to test systematically some
+# functionalities and invariance, e.g. SYMMETRIC_METRICS lists all metrics that
+# are symmetric with respect to their input argument y_true and y_pred.
+#
+
+# Symmetric with respect to their input arguments y_true and y_pred
+# metric(y_true, y_pred) == metric(y_pred, y_true).
+SYMMETRIC_METRICS = [ "adjusted_rand_score" , "v_measure_score" ,
+    "mutual_info_score","adjusted_mutual_info_score",
+    "normalized_mutual_info_score",
+    ]
+                    
+NON_SYMMETRIC_METRICS = ["homogeneity_score" , "completeness_score"]
+                    
+#METIRICS with output between 0 and 1 and belong to BICLUSTER METRICS
+METRICS_NORMALIZED_OUTPUT_BICLUSTER = ["consensus_score","jaccard"]
+ 
+#metrics with output between 0 and 1                                      
+METRICS_NORMALIZED_OUTPUT = ["adjusted_rand_score","homogeneity_score",
+    "completeness_score","v_measure_score",
+    "adjusted_mutual_info_score","normalized_mutual_info_score"
+    ]                  
+                                
+#when information is zero these metrics output zero
+METRICS_ZERO_INFO = ["normalized_mutual_info_score","v_measure_score",
+    "adjusted_mutual_info_score"
+    ]
+                      
+#METRICS where permutations oflabels dont change score
+METRICS_PERMUTE_LABELS = ["homogeneity_score","v_measure_score",
+    "completeness_score","mutual_info_score","adjusted_mutual_info_score",
+    "normalized_mutual_info_score"
+    ]
+                        
+#metrics which result in 0 when a class is split across different clusters
+ClASS_BASED_METRICS = ["adjusted_mutual_info","normalized_mutual_info",
+    "v_measure_score"
+    ]
+                        
+
+def test_symmetry():
+    #symmetric_metrics
+    for name in SYMMETRIC_METRICS:
+        metric=ALL_METRICS[name]
+        assert_almost_equal(metric([0,1,0],[42,7,42]),

```

Comment:
You can also use something like `y3 = y2.copy(); y3[::3] = 3 - y3[::3]` to make a set of labels that differs from y2 in only 1/3 of instances.